public/app/features/inspector/InspectJSONTab.tsx

```diff
@@ -121,6 +122,14 @@ export class InspectJSONTab extends PureComponent<Props, State> {
         } else {
           const updates = JSON.parse(this.state.text);
           dashboard!.shouldUpdateDashboardPanelFromJSON(updates, panel!);
+
+          //Report relevant updates
+          reportInteraction('grafana_panel_inspect_applied_clicked', {
+            panel_type_changed: panel!.type !== updates.type,
+            panel_id_changed: panel!.id !== updates.id,
+            panel_grid_pos_changed: !isEqual(panel!.gridPos, updates.gridPos),
```

Comment:
Thinking about the main usage I may have on editing the JSON directly, I can think on:
> Find and replace for queries: When I want to change multiple queries to point to another SQL table, for example, I probably would go to edit the JSON instead to go to the UI

Do you think we can check this somehow? I see it hard because of the nature of the data sources, each of them has a different shape 😓

Reply:
IMO with this PR we are trying to figure out _if_, not _how_ the Apply button is used. 

What I think is relevant from that standpoint is learning "How many panel inspect interactions resulted in Apply button clicked". And to do that we need to correlate the source and target event by some kind of a key passed as an interaction identifier. In my view we need the following interactions reported:

- `grafana_panel_inspect_opened{interactionId: 'XXX'}`
- `grafana_panel_inspect_apply_clicked{interactionId: 'XXX'}`
- `grafana_panel_inspect_closed{interactionId: 'XXX'}`
 
Not sure `interactionId` is the name to use, but it should be passed as the event payload.

Reporting the apply clicked only will give use 100% of the usages that cannot be analyzed in any context.

Reply:
Great suggestions @dprokop 😄 , I think adding the `grafana_panel_inspect_opened` event will do the job.

With that event and the  `grafana_panel_inspect_applied_clicked` event we can get the percentage rate of `how many people opened the "Panel Inspect ->Json"`  vs `how many hit the "Apply" button`, this will answer the question: "Is the Apply button used?" 

I don't think is necessary to add `interactionId` or the `grafana_panel_inspect_closed` to get that percentage rate.

Reply:
@dprokop do you think we need an identifier for this funnel? 

Since we can only access to "apply" button while displaying inspect we probably don't need to add the interaction since we can do: `%_applied_changes = total_inspect_views / apply_button_clicks`. Am I missing something?

I see the identifier helpful when we're measuring funnels that might differ, for example, `Open Dashboard -> Edit panel -> Save dashboard`. You can save the dashboard for so many reasons, so you want to make sure you save it because you have edited a panel.

Reply:
Why do you think 
>Reporting the apply clicked only will give use 100% of the usages that cannot be analyzed in any context.

If the idea of this PR is to collect information how many users are trying to make a change on those properties that should not be manually changed, then we only want to follow the real attempts, not also looking into the Panel JSON without applying anything to it?

In general to me it seems this PR is okay with what it has, maybe except the `panel_grid_pos_changed` because I don't see its value to the idea right now.

If we want to later also follow user's interaction or know which properties are being changed out of the ones that are supposed to be changed, I think it's better to rename `grafana_panel_inspect_applied_clicked` to something more error-specific like `grafana_panel_inspect_forbidden_changes_applied` (or better).

Reply:
> Since we can only access to "apply" button while displaying inspect we probably don't need to add the interaction since we can do: %_applied_changes = total_inspect_views / apply_button_clicks. Am I missing something?

That's my point, we don't have `total_inspect_views`, unless I don't know about sth :)